clang/lib/Driver/Driver.cpp

```diff
@@ -4699,15 +4718,12 @@ class OffloadingActionBuilder final {
     }
 
     void addDeviceLinkDependencies(OffloadDepsJobAction *DA) override {
-      for (unsigned I = 0; I < ToolChains.size(); ++I) {
-        // Register dependent toolchain.
-        withBoundArchForToolChain(ToolChains[I], [&](const char *BoundArch) {
-          DA->registerDependentActionInfo(ToolChains[I], BoundArch,
-                                          Action::OFK_SYCL);
-        });
-
-        // Add deps output to linker inputs.
+      unsigned I = 0;
+      for (auto &TargetInfo : SYCLTargetInfoList) {
+        DA->registerDependentActionInfo(TargetInfo.TC, TargetInfo.BoundArch,
+                                        Action::OFK_SYCL);
         DeviceLinkerInputs[I].push_back(DA);
+        I++;
```

Comment:
One could get fancy here, but I'm not sure as to what's more readable
```suggestion
        DeviceLinkerInputs[I++].push_back(DA);
```